apps/web/src/lib/utils.ts

```diff
@@ -43,3 +50,129 @@ export const insertSoftHyphens = (text: string): string => {
     .replaceAll(/(?<word>@)/g, "\u200b$<word>")
     .replaceAll(/(?<word>toimikunta)/g, "\u00ad$<word>");
 };
+
+/**
+ * Get date in format 12.2.2024 18:00:03
+ */
+export const formatDateTimeSeconds = (
+  date: string,
+  locale?: Locale,
+): string => {
+  const formatter = new Intl.DateTimeFormat(`${locale ?? "fi"}-FI`, {
+    day: "numeric",
+    month: "numeric",
+    year: "numeric",
+    hour: "numeric",
+    minute: "numeric",
+    second: "numeric",
+  });
+
+  return formatter.format(new Date(date));
+};
+
+/**
+ * Get date in format "su 12.2. klo 18"
+ */
+export const formatDatetime = (date: string, locale?: Locale): string => {
+  const formatter = new Intl.DateTimeFormat(`${locale ?? "fi"}-FI`, {
+    weekday: "short",
+    day: "numeric",
+    month: "numeric",
+    hour: "numeric",
+    minute: "numeric",
+  });
+
+  return formatter.format(new Date(date));
+};
+
+/**
+ * Get date in format "su 12.2.2024 klo 18"
+ */
+export const formatDatetimeYear = (date: string, locale?: Locale): string => {
+  const formatter = new Intl.DateTimeFormat(`${locale ?? "fi"}-FI`, {
+    weekday: "short",
+    day: "numeric",
+    month: "numeric",
+    year: "numeric",
+    hour: "numeric",
+    minute: "numeric",
+  });
+
+  return formatter.format(new Date(date));
+};
+
+/**
+ * Get date in format "12.2.2024"
+ */
+export const formatDate = (date: string, locale?: Locale): string => {
+  const formatter = new Intl.DateTimeFormat(`${locale ?? "fi"}-FI`, {
+    day: "numeric",
+    month: "numeric",
+    year: "numeric",
+  });
+
+  return formatter.format(new Date(date));
+};
+
+export const getQuotasWithOpenAndQueue = (
```

Comment:
I think this stuff is also available in `@tietokilta/ilmomasiina-components` - might be split in the future to provide such utility code without any UI deps

Reply:
Yeah, there was something but not exactly the same. Might make sense to unify later but not a big deal for such small functionality